src/Helper/ObjectParser.php

```diff
@@ -0,0 +1,167 @@
+{
+    /**
+     * @psalm-var array<string, array<string, array>>
+     */
+    #[ArrayShape([
+        [
+            'rules' => 'array',
+            'reflectionAttributes' => 'array',
+        ],
+    ])]
+    private static array $cache = [];
+    private string|null $cacheKey;
+
+    public function __construct(
+        private object $object,
+        private int $propertyVisibility = ReflectionProperty::IS_PRIVATE |
+        ReflectionProperty::IS_PROTECTED |
+        ReflectionProperty::IS_PUBLIC,
+        private bool $skipStaticProperties = false,
+        bool $useCache = true
+    ) {
+        $this->cacheKey = $useCache
+            ? $this->object::class . '_' . $this->propertyVisibility . '_' . $this->skipStaticProperties
+            : null;
+    }
+
+    /**
+     * @return array<string, list<RuleInterface>>
+     */
+    public function getRules(): array
+    {
+        if ($this->hasCacheItem('rules')) {
+            /** @var array<string, list<RuleInterface>> */
+            return $this->getCacheItem('rules');
+        }
+
+        $rules = [];
+        foreach ($this->getReflectionProperties() as $property) {
+            // TODO: use Generator to collect attributes.
+            $attributes = $property->getAttributes(RuleInterface::class, ReflectionAttribute::IS_INSTANCEOF);
+            foreach ($attributes as $attribute) {
+                $rule = $attribute->newInstance();
+                $rules[$property->getName()][] = $rule;
+
+                if ($rule instanceof AfterInitAttributeEventInterface) {
+                    $rule->afterInitAttribute($this->object);
+                }
+            }
+        }
+
+        if ($this->useCache()) {
+            $this->setCacheItem('rules', $rules);
+        }
+
+        return $rules;
+    }
+
+    public function getAttributeValue(string $attribute): mixed
+    {
+        return ($this->getReflectionProperties()[$attribute] ?? null)?->getValue($this->object);
+    }
+
+    public function hasAttribute(string $attribute): bool
+    {
+        return array_key_exists($attribute, $this->getReflectionProperties());
+    }
+
+    public function getData(): array
+    {
+        $data = [];
+        foreach ($this->getReflectionProperties() as $name => $property) {
+            /** @var mixed */
+            $data[$name] = $property->getValue($this->object);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @return array<string, ReflectionProperty>
+     */
+    public function getReflectionProperties(): array
+    {
+        if ($this->hasCacheItem('reflectionProperties')) {
+            /** @var array<string, ReflectionProperty> */
+            return $this->getCacheItem('reflectionProperties');
+        }
+
+        $reflection = new ReflectionObject($this->object);
+        $reflectionProperties = [];
+
+        foreach ($reflection->getProperties($this->propertyVisibility) as $property) {
+            if ($this->skipStaticProperties && $property->isStatic()) {
```

Comment:
Could you add test to check skipping of static properties?

Reply:
done